librz/analysis/class.c

```diff
@@ -950,6 +950,15 @@ RZ_API RzAnalysisClassErr rz_analysis_class_vtable_get(RzAnalysis *analysis, con
 		vtable->size = 0;
 	}
 
+	RVTableContext context;
+	ut64 startAddress = vtable->addr;
+	rz_analysis_vtable_begin(analysis, &context);
+	RVTableInfo *vtablet = rz_analysis_vtable_parse_at(&context, startAddress);
```

Comment:
This is not the right place to approach this. `rz_analysis_class_vtable_get()` is a pure getter function, which should only retrieve information stored in the database, but `rz_analysis_vtable_begin()` etc. are actual analysis passes.

Instead, what you want to do is fix the analysis that writes to the database in the first place, which in this case should be here:
https://github.com/rizinorg/rizin/blob/b213e62beef770fed766f65e8548dffd27501ce2/librz/analysis/rtti_itanium.c#L765
You can see it sets the size to 0, which should be replaced by the correct size.

Reply:
I get it, gonna set about it and understand the whole structure better